docs/development/module/tests.md

```diff
@@ -6,126 +6,123 @@ Runtime tests allow you to verify the logic in your runtime module by mocking a
 
-You can set custom values for the genesis configuration for modules you include in your mock runtime.
+#### Genesis Config
 
-TODO
+The previously shown `ExtBuilder::build()` method used the default genesis configuration for building the mock runtime environment. In many cases, it is convenient to set storage before testing.
 
-### Block Production
+An example might involve pre-seeding account balances before testing.
+
+In the implementation of `system::Trait`, `AccountId` is set to `u64` just like `Balance` shown above. Place `(u64, u64)` pairs in the `balances` vec to seed `(AccountId, Balance)` pairs as the account balances.
+
+```rust, ignore
+pub fn build(self) -> runtime_io::TestExternalities {
+	GenesisConfig {
+		balances: Some(balances::GenesisConfig::<TestRuntime>{
+			balances: vec![
+				(1, 10),
+				(2, 20),
+				(3, 30),
+				(4, 40),
+				(5, 50),
+				(6, 60)
+			],
+			vesting: vec![],
+		}),
+	}.build_storage().unwrap().into()
+}
+```
+
+Account 1 has balance 10, account 2 has balance 20, and so on.
 
-Your tests may require you to simulate block production to move forward the state of your chain.
+### Block Production
 
-You will need to emulate the orchestration provided by the Executive module in order for your runtime to behave correctly.
+It will be useful to simulate block production to verify that expected behavior holds during block time dependent changes. 
 
-Usually this means executing the `on_initialize` and `on_finalize` functions of all the modules included in your test, and incrementing the block number tracked by the System module. This can be written as a function you include in your unit tests:
+A simple way of doing this increments the System module's block number between `on_initialize` and `on_finalize` calls from all modules with `System::block_number()` as the sole input. While it is important for runtime code to [cache calls](https://substrate.dev/recipes/storage/cache.html) to storage or the system module, the test environment scaffolding should prioritize readability to facilitate future maintenance.
 
-```rust
+```rust, ignore
 fn run_to_block(n: u64) {
 	while System::block_number() < n {
-		MyModule::on_finalize(System::block_number());
+		ExampleModule::on_finalize(System::block_number());
 		System::on_finalize(System::block_number());
 		System::set_block_number(System::block_number() + 1);
 		System::on_initialize(System::block_number());
-		MyModule::on_initialize(System::block_number());
+		ExampleModule::on_initialize(System::block_number());
 	}
 }
 ```
 
-You can then use this function in your unit tests:
+`on_finalize` and `on_initialize` are only called from `ExampleModule` if the pallet's trait implements the `sr_primitives::traits::{OnInitialize, OnFinalize}` traits to execute the logic encoded in the runtime methods before and after each block respectively.
+
+To use this function in unit tests,
 
-```rust
+```rust, ignore
```

Comment:
Why are you adding `ignore` to this doc? It only has meaning in RustDocs right?

Reply:
deleted and IDK, it also has meaning in mdbook